bot/sx-bot.el

```diff
@@ -0,0 +1,82 @@
+;;; sx-bot.el --- Functions for automated maintanence  -*- lexical-binding: t; -*-
+
+;; Copyright (C) 2014  Artur Malabarba
+
+;; Author: Artur Malabarba <bruce.connor.am@gmail.com>
+
+;; This program is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+;;; Commentary:
+
+;; This file defines the behavior of a bot.  To allow completion for
+;; tags, this bot runs through all sites in the network and retrieves
+;; all of their tags.  This data is then written to a directory which
+;; is tracked by the git repository.
+
+
+;;; Code:
+
+(require 'package)
+(package-initialize)
+
+(require 'sx-load)
+
+(setq sx-request-remaining-api-requests-message-threshold 50000)
+
+(defcustom sx-bot-out-dir "./data/tags/"
+  "Directory where output tag files are saved."
+  :type 'directory
+  :group 'sx)
+
+
+;;; Printing
+(defun sx-bot-write-to-file (data)
+  "Write (cdr DATA) to file named (car DATA).
+File is savedd in `sx-bot-out-dir'."
+  (let ((file-name (expand-file-name (car data) sx-bot-out-dir)))
+    (with-temp-file file-name
+      (let* (print-length
+             (repr (prin1-to-string
```

Comment:
Agreed. 
One more thing. How much do we care about optimising download speed here? 

- We could zip the tag files.
- we could also write the tag names as symbols instead of strings, thus reducing two characters per tag (the double quotes). But this might just be silly. 

OTOH, if I remember correctly you said the entire thing was very short, so even zipping the files might not make any relevant difference. 
It might make a difference if we ever decide to provide the tag excerpts along with the names.